technical-reports/format/groups.md

```diff
@@ -164,6 +158,46 @@ For example:
 
 </aside>
 
+### Extensions
+
+Groups MAY include an optional `$extensions` property where tools MAY add proprietary, user-, team- or vendor-specific data to a group. When doing so, each tool MUST use a vendor-specific key whose value MAY be any valid JSON data.
+
+Group extensions follow the same rules as [design token extensions](#extensions), the only difference is that they relate to the group itself.
+
+- The keys SHOULD be chosen such that they avoid the likelihood of a naming clash with another vendor's data. The [reverse domain name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) is recommended for this purpose.
+- Tools that process design token files MUST preserve any extension data they do not themselves understand. For example, if a group contains extension data from tool A and the file containing that data is opened by tool B, then tool B MUST include the original tool A extension data whenever it saves a new design token file containing that group.
+
+Note that, since a group's `$extensions` only relate to that group, they do not have any effect on nested groups or tokens.
```

Comment:
I'm wondering whether this sentence is straightforward enough. 

Some parts of #106 feels a bit more explicit (too much?)
> Groups `$extensions` should not be inherited. i.e. tokens and groups within a group that has `$extensions` should not be treated as though they themselves had the same $extensions property.